LoRaEngine/test/LoRaWan.Test.Shared/IntegrationTestFixtureBase.Asserts.cs

```diff
@@ -135,6 +135,52 @@ public async Task AssertNetworkServerModuleLogExistsAsync(Func<string, bool> pre
             }
         }
 
+        /// <summary>
+        /// Check for The timing between an upstream message and a downstream one.
+        /// The method stop at first occurence found and is not expected to perform against multiple DS/US messages.
+        /// </summary>
+        public async Task CheckAnswerTimingAsync(uint rxDelay, bool isSecondWindows)
+        {
+            var upstreamMessageTimingResult = await this.TryFindMessageTimeAsync("rxpk");
+            var downstreamMessageTimingResult = await this.TryFindMessageTimeAsync("txpk");
+
+            if (upstreamMessageTimingResult.Success && upstreamMessageTimingResult.Success)
+            {
+                Assert.Equal(rxDelay + (isSecondWindows ? 1000000 : 0), downstreamMessageTimingResult.Result - upstreamMessageTimingResult.Result);
+            }
+            else
+            {
+                Assert.True(false, "Could not procees with timestamp message search.");
+            }
+        }
+
+        /// <summary>
+        /// Helper method to find the time of the message that contain the message argument.
+        /// </summary>
+        async Task<FindTimeResult> TryFindMessageTimeAsync(string message)
+        {
+            var log = await this.SearchUdpLogs(x => x.Contains(message));
+            int timeIndexStart = log.FoundLogResult.IndexOf(@"""tmst"":") + 7;
```

Comment:
+7 is likely just moving to the end of the token. What I usually do here, is create a const token (const string token = @"""tmst"":") and then use the token.Length to make it obvious.

Reply:
I could do this in part, but I can't use the token.length as my string contains escape characters...

Reply:
The length of the string should be correct no matter if you are using escape chars or not. If you put this in a const and check the Length, it will be 7.